nodes.py

```diff
@@ -272,7 +258,59 @@ def nodefor(self, element):
             "ulink": UlinkNode,
             "verbatim": VerbatimNode,
             None: Node,
-        }[element.name]
+        }
+        if element.name == "compounddef":
+            if element["kind"] == "page":
+                return defaults[element.name]
+            else:
+                return {
+                    "group": GroupNode,
+                }[element["kind"]]
```

Comment:
Is this equivalent to the following:

```suggestion
        if element.name == "compounddef":
            return {
                "group": GroupNode,
                "page": SectNode,
            }[element["kind"]]
```